react-blog-app/src/App.tsx

```diff
@@ -1,34 +1,49 @@
-import React, { useState } from "react";
+import React from "react";
+import { Switch, Route, Redirect, useLocation } from "react-router-dom";
 import "./App.css";
 
 import Header from "./components/Header";
 import Footer from "./components/Footer";
+import Modal from "./components/Modal";
 import MainPage from "./pages/MainPage";
-import NewArticleForm from "./pages/NewArticleForm";
-import FormSuccess from "./pages/FormSuccess";
+import ArticlePage from "./pages/ArticlePage";
+import NewArticlePage from "./pages/NewArticlePage";
+import NotFoundPage from "./pages/NotFoundPage";
+import NewArticleModal from "./modals/NewArticleModal";
+import FormSuccessModal from "./modals/FormSuccessModal";
 
 const App = () => {
 
-    const [isNewArticleOpen, setIsNewArticleOpen] = useState(false);
-    const [isNewArticleSubmitted, setIsNewArticleSubmitted] = useState(false);
+    const location = useLocation<any>();
+    const background = location.state && location.state.background;
 
     return (
         <div className="app">
-            {
-                isNewArticleOpen && !isNewArticleSubmitted &&
-                <NewArticleForm
-                    setOpenClosed={setIsNewArticleOpen}
-                    setIsSubmitted={setIsNewArticleSubmitted}
-                />
-            }
-            {
-                isNewArticleSubmitted && !isNewArticleOpen &&
-                <FormSuccess
-                    setOpenClosed={setIsNewArticleSubmitted}
-                />
-            }
-            <Header  setIsNewArticleOpen={setIsNewArticleOpen} />
-            <MainPage isNewArticleSubmitted={isNewArticleSubmitted} />
+            <Header />
+
+            {/* Routing */}
+            <Switch location={ background || location }>
+                <Route exact path="/" component={MainPage} />
+                <Route exact path="/newarticle" component={NewArticlePage} />
+                <Route path="/article_:articleId" component={ArticlePage} />
+                <Route path="/404" component={NotFoundPage} />
+                <Route path="*">
+                    <Redirect to="/404"/>
+                </Route>
+            </Switch>
+
+            {background && background.pathname !== "/newarticle/success" && <Route exact path="/newarticle/success" children={
```

Comment:
it's probably not a good practice to create a separate state for the success message
it's usually just a popup that is shown and then hidden, we don't need to share that state, we even shouldn't do that

Reply:
Got it! Thank you!